jetbrains-core/src/software/aws/toolkits/jetbrains/services/cloudformation/CfnFileTypeDetector.kt

```diff
@@ -0,0 +1,39 @@
+// Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.jetbrains.services.cloudformation
+
+import com.intellij.openapi.fileTypes.FileType
+import com.intellij.openapi.fileTypes.FileTypeRegistry
+import com.intellij.openapi.util.io.ByteSequence
+import com.intellij.openapi.vfs.VirtualFile
+import software.aws.toolkits.jetbrains.services.cloudformation.json.JSON_FILE_TYPE_NAME
+import software.aws.toolkits.jetbrains.services.cloudformation.yaml.YAML_FILE_TYPE_NAME
+
+class CloudFormationFileTypeDetector : FileTypeRegistry.FileTypeDetector {
+    override fun getVersion(): Int = 1
+
+    override fun detect(file: VirtualFile, firstBytes: ByteSequence, firstCharsIfText: CharSequence?): FileType? {
+        if (firstCharsIfText == null) {
+            return null
+        }
+
+        if (file.extension != "template") {
+            return null
+        }
+
+        if (SERVERLESS_TRANSFORM in firstCharsIfText || CFN_FORMAT_VERSION in firstCharsIfText) {
+            val firstChar = firstCharsIfText.first { !it.isWhitespace() }
+
+            // Note: We find by name in case it is not loaded due to upstream plugin (yaml/json) is not enabled
+            val registry = FileTypeRegistry.getInstance()
+            return if (firstChar == '{') {
```

Comment:
I know this was the existing detection code but `[`, is also valid JSON (despite being invalid CFN template) so maybe we should do the same check we do in VSTS https://github.com/aws/aws-vsts-tools/pull/332/files#diff-410117ffb18d53f752661b2ac5e3e65cR62

Reply:
But very invalid CloudFrormation, its not meant to identify JSON or not, its to identify if its YAML CloudFromation or JSON Cloudformation

Reply:
Fair, makes sense